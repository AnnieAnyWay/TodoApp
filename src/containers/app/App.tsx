import React from "react";
import TodoList from "../../components/todo/todo-list.block";
import Header from "../../components/header/header.block";

function App() {
  return (
    <div>
      <Header />
      <TodoList />
    </div>
  );
}

export default App;
