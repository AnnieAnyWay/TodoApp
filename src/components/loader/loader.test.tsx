import React from "react";
import renderer from "react-test-renderer";
import { combineReducers, createStore, compose, applyMiddleware } from "redux";
import thunkMiddleware from "redux-thunk";
import rootReducer, { initialState } from "../../store/reducers";

import { Provider } from "react-redux";
import App from "../../containers/app/App";
import Loader from "./loader.block";

function create(state = {}) {
  return createStore(
    rootReducer,
    {
      ...initialState,
      ...state,
    },
    compose(applyMiddleware(thunkMiddleware))
  );
}

const mockedLoader = {
  todos: [],
  ui: { isLoading: true },
};

it("renders loader", () => {
  const store = create(mockedLoader);
  const snap = renderer
    .create(
      <Provider store={store}>
        <Loader />
      </Provider>
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
