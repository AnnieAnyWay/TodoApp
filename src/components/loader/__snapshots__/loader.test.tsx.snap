// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loader 1`] = `
<div
  className="loader-container"
>
  <div
    className="loader"
  >
    <div />
    <div />
  </div>
</div>
`;
